File: 1.Fundamental-Level/3.OOP/04.Inheritance-and-Abstraction-Homework/2.Animals/Animal.cs
using _2.Animals.Interfaces;

public abstract class Animal : ISoundProducible
{
    public Animal(string name, int age, string gender)
    {
        this.Name = name;
        this.Age = age;
        this.Gender = gender;
    }

    public string Name { get; set; }

    public int Age { get; set; }

    public string Gender { get; set; }

    public abstract void ProduceSound();
}

File: Java-Development/Java-OOP-Basics/Java-OOP-Basics-Solutions/src/polymorphism/_03_WildFarm/animals/Animal.java
package polymorphism._03_WildFarm.animals;

import polymorphism._03_WildFarm.foods.Food;

public abstract class Animal {

    private String animalName;
    private Double animalWeight;
    private Integer foodEaten;

    public Animal(String name, Double weight) {
        this.animalName = name;
        this.animalWeight = weight;
        this.foodEaten = 0;
    }

    protected void setFoodEaten(Integer foodEaten) {
        this.foodEaten = foodEaten;
    }

    protected Integer getFoodEaten() {
        return foodEaten;
    }

    protected String getAnimalName() {
        return animalName;
    }

    protected Double getAnimalWeight() {
        return animalWeight;
    }

    public abstract void makeSound();

    public void eat(Food food) {
        this.foodEaten += food.getQuantity();
    }
}